chebi-update/src/main/java/org/reactome/release/chebiupdate/ChebiDataRetriever.java

```diff
@@ -0,0 +1,264 @@
+					{
+						Entity entity;
+						// only query web service if the data is not in the chebi-cache - NOTE: chebiCache will always be empty if this.useCache == false
+						if (!chebiCache.containsKey("CHEBI:"+identifier))
+						{
+							entity = this.getChEBIDataFromWebService(failedEntitiesList, chebiCache, bw, molecule);
+						}
+						else // ...Load data from the cache.
+						{
+							entity = this.extractChEBIEntityFromCache(chebiCache, identifier);
+						}
+						// Add entity to map (if it's non-null)
+						if (entity != null)
+						{
+							entityMap.put(molecule.getDBID(), entity);
+						}
+					}
+					else
+					{
+						logger.error("ERROR: Instance \"{}\" has an empty/null identifier. This should not be allowed.", molecule.toString());
+						failedEntitiesList.put(molecule, molecule.toString() + " has an empty/NULL identifier.");
+					}
+					int i = counter.getAndIncrement();
+					if (i % 250 == 0)
+					{
+						logger.debug("{} ChEBI identifiers checked", i);
+					}
+				}
+				catch (ChebiWebServiceFault_Exception e)
+				{
+					ChebiDataRetriever.handleWSException(failedEntitiesList, molecule, identifier, e);
+				}
+				catch (InvalidAttributeException e)
+				{
+					logger.error("InvalidAttributeException caught while trying to get the \"identifier\" attribute on " + molecule.toString());
+					// stack trace should be printed, but I don't think this should break execution, though the only way I can think
+					// of this happening is if the data model changes - otherwise, this exception would probably never be caught.
+					e.printStackTrace();
+				}
+				catch (Exception e)
+				{
+					// general exceptions - print stack trace but keep going.
+					e.printStackTrace();
+				}
+			});
+		}
+		return entityMap;
+	}
+
+	/**
+	 * Handles exceptions from the ChEBI web service. Some of them are ok, as as "invalid ChEBI identifier",
+	 * and "entity is deleted, obsolete, not released", but anything else
+	 * will cause a runtime exception to be thrown. It's probably not safe to continue with unrecognized exceptions.
+	 * IF you encounter *new* exceptions that are not listed here, but are not too serious, feel free to update this code
+	 * to write the appropriate message for those exceptions, rather than failing with a RuntimeException.
+	 * @param failedEntitiesList
+	 * @param molecule
+	 * @param identifier
+	 * @param e
+	 */
+	private static void handleWSException(Map<GKInstance, String> failedEntitiesList, GKInstance molecule, String identifier, ChebiWebServiceFault_Exception e)
+	{
+		// "invalid ChEBI identifier" probably shouldn't break execution but should be logged for further investigation.
+		if (e.getMessage().contains("invalid ChEBI identifier"))
+		{
+			logger.error("ERROR: ChEBI Identifier \"{}\" is not formatted correctly.", identifier);
+			failedEntitiesList.put(molecule, "ChEBI Identifier \""+identifier+"\" is not formatted correctly.");
```

Comment:
Could the common error message be extracted into a temp and used in the logger and failedEntitiesList here.   That way it only needs to be edited in one place.  The same for the next branch as well.